packages/loader/container-loader/src/container.ts

```diff
@@ -1642,6 +1686,19 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         } else if (value === ConnectionState.Disconnected) {
             // Important as we process our own joinSession message through delta request
             this.pendingClientId = undefined;
+            // Only wait for "leave" message if we have some outstanding ops and the client was write client as
+            // server would not accept ops from read client.
+            if (this._deltaManager.clientSequenceNumber - this._deltaManager.clientSequenceNumberObserved > 0
+                && this.client.mode === "write"
+            ) {
+                this.prevClientLeftP = new Deferred();
+                // Max time is 5 min for which we are going to wait for its own "leave" message.
+                setTimeout(() => {
+                    if (this.prevClientLeftP?.isCompleted === false) {
```

Comment:
There is no need to check for isCompleted (and usually it's not good practice) - resolving same promise again is Ok and is noop